### PR TITLE
fix: only select current node on triple click

### LINKED ISSUE #27: Triple-click selects the entire text in Markdown mode
Reported on [Meta.SE](https://meta.stackexchange.com/a/360115/395497).

**Describe the bug**

> It's a standard practice in modern text fields to double-click in order to select a single word and triple-click to select a single **paragraph**. This has worked in the old editor, works in the Rich Text editor, but does not work in Markdown mode and the entire text gets selected instead.
> 
> Selecting paragraph in Rich Text vs Markdown mode

**Expected behavior**

The paragraph the cursor is currently in is selected, not the entirety of the text.

### PR DESCRIPTION
**Pull request by dancormier**
fixes #27

Note: this solution was largely lifted from [a comment on the ProseMirror discussion forum](https://discuss.prosemirror.net/t/how-can-i-let-the-browser-instead-of-prosemirror-view-handle-triple-clicks/4526/3).

<details>
<summary>Before</summary>

</details>

<details>
<summary>After</summary>

</details>

---

> **Note**
> Just now seeing #109! Great minds think alike.

### CODE REVIEW

File: src/commonmark/plugins/triple-click-handler.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Plugin } from "prosemirror-state";
+
+/** Plugin that selects only the current node on triple click */
```
Comment (b-kelly) [inline src/commonmark/plugins/triple-click-handler.ts:3]: The doc here is slightly misleading:

```suggestion
/** Plugin that restores browser default triple-click handling behavior */
```

**Review by b-kelly — CHANGES_REQUESTED** [review]
As mentioned on #109:
> I don't think this is a proper fix to the root issue, but rather fixing one of the symptoms. I've tackled the root cause here a few times, but have yet to implement a proper fix

That being said, the fix for this issue is pretty elusive, so this may work in the meanwhile. I'd _really_ like some tests for this. I agree that unit tests may not be feasible, but we can at least get some e2e tests in.

**Comment by dancormier** [comment]
> …I'd _really_ like some tests for this. I agree that unit tests may not be feasible, but we can at least get some e2e tests in.

You got it. I added [E2E tests for triple click and double click](https://github.com/StackExchange/Stacks-Editor/pull/128/commits/9a5e9a697fb415be742161ff1e365533a01e6e6d) for good measure.

**Comment by b-kelly** [comment]
Getting failing tests locally:

```
  1) [chromium] › commonmark\editor.e2e.test.ts:33:5 › markdown mode › should select word on double click

    Timeout of 30000ms exceeded.

    Error: expect(received).toEqual(expected) // deep equality

    Expected: "paragraph"
    Received: "paragraph "

      38 |             window.getSelection().toString()
      39 |         );
    > 40 |         expect(selectedText).toEqual("paragraph");
         |                              ^
      41 |     });
      42 |
      43 |     test("should select line on triple click", async () => {

        at C:\code\Stacks-Editor\test\commonmark\editor.e2e.test.ts:40:30
```
```
  3 failed
    [chromium] › commonmark\editor.e2e.test.ts:33:5 › markdown mode › should select word on double click
    [firefox] › commonmark\editor.e2e.test.ts:33:5 › markdown mode › should select word on double click
    [webkit] › commonmark\editor.e2e.test.ts:33:5 › markdown mode › should select word on double click
```

**Comment by dancormier** [comment]
@b-kelly That's very odd! Is there a chance that triple click operates differently on Windows than Mac OS?

I'm having some weird E2E test issues locally, but if I run the commonmark tests directly, all they pass 🤷‍♂️

```
npx playwright test test/commonmark/editor.e2e.test.ts 

Running 5 tests using 1 worker

  ✓  test/commonmark/editor.e2e.test.ts:23:5 › markdown mode › should show toggle switch (2s)
  ✓  test/commonmark/editor.e2e.test.ts:29:5 › markdown mode › should render menu bar (18ms)
  ✓  test/commonmark/editor.e2e.test.ts:33:5 › markdown mode › should select word on double click (129ms)
  ✓  test/commonmark/editor.e2e.test.ts:43:5 › markdown mode › should select line on triple click (142ms)
  ✓  test/commonmark/editor.e2e.test.ts:56:5 › markdown mode › should not highlight bold menu button after click (114ms)


  5 passed (3s)
```

**Pull request merged**